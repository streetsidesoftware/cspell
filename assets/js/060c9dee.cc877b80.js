"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[94023],{46039:(e,t,n)=>{n.d(t,{R:()=>l,x:()=>c});var i=n(18225);const s={},r=i.createContext(s);function l(e){const t=i.useContext(r);return i.useMemo((function(){return"function"==typeof e?e(t):{...t,...e}}),[t,e])}function c(e){let t;return t=e.disableParentContext?"function"==typeof e.components?e.components(s):e.components||s:l(e.components),i.createElement(r.Provider,{value:t},e.children)}},87257:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>a,contentTitle:()=>c,default:()=>p,frontMatter:()=>l,metadata:()=>i,toc:()=>o});const i=JSON.parse('{"id":"api/cspell-lib/functions/readRawSettings","title":"Function: readRawSettings()","description":"readRawSettings(filename, relativeTo?): Promise\\\\","source":"@site/docs/api/cspell-lib/functions/readRawSettings.md","sourceDirName":"api/cspell-lib/functions","slug":"/api/cspell-lib/functions/readRawSettings","permalink":"/docs/api/cspell-lib/functions/readRawSettings","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell-lib/functions/readRawSettings.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Function: readFileSync()","permalink":"/docs/api/cspell-lib/functions/readFileSync"},"next":{"title":"Function: readSettings()","permalink":"/docs/api/cspell-lib/functions/readSettings"}}');var s=n(37557),r=n(46039);const l={},c="Function: readRawSettings()",a={},o=[{value:"Parameters",id:"parameters",level:2},{value:"filename",id:"filename",level:3},{value:"relativeTo?",id:"relativeto",level:3},{value:"Returns",id:"returns",level:2}];function d(e){const t={a:"a",blockquote:"blockquote",code:"code",h1:"h1",h2:"h2",h3:"h3",header:"header",p:"p",strong:"strong",...(0,r.R)(),...e.components};return(0,s.jsxs)(s.Fragment,{children:[(0,s.jsx)(t.header,{children:(0,s.jsx)(t.h1,{id:"function-readrawsettings",children:"Function: readRawSettings()"})}),"\n",(0,s.jsxs)(t.blockquote,{children:["\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.strong,{children:"readRawSettings"}),"(",(0,s.jsx)(t.code,{children:"filename"}),", ",(0,s.jsx)(t.code,{children:"relativeTo"}),"?): ",(0,s.jsx)(t.code,{children:"Promise"}),"<",(0,s.jsx)(t.a,{href:"/docs/api/cspell-lib/interfaces/CSpellSettingsWithSourceTrace",children:(0,s.jsx)(t.code,{children:"CSpellSettingsWithSourceTrace"})}),">"]}),"\n"]}),"\n",(0,s.jsxs)(t.p,{children:["Defined in: ",(0,s.jsx)(t.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell-lib/src/lib/Settings/Controller/configLoader/defaultConfigLoader.ts#L99",children:"packages/cspell-lib/src/lib/Settings/Controller/configLoader/defaultConfigLoader.ts:99"})]}),"\n",(0,s.jsx)(t.h2,{id:"parameters",children:"Parameters"}),"\n",(0,s.jsx)(t.h3,{id:"filename",children:"filename"}),"\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.code,{children:"string"})," | ",(0,s.jsx)(t.code,{children:"URL"})]}),"\n",(0,s.jsx)(t.h3,{id:"relativeto",children:"relativeTo?"}),"\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.code,{children:"string"})," | ",(0,s.jsx)(t.code,{children:"URL"})]}),"\n",(0,s.jsx)(t.h2,{id:"returns",children:"Returns"}),"\n",(0,s.jsxs)(t.p,{children:[(0,s.jsx)(t.code,{children:"Promise"}),"<",(0,s.jsx)(t.a,{href:"/docs/api/cspell-lib/interfaces/CSpellSettingsWithSourceTrace",children:(0,s.jsx)(t.code,{children:"CSpellSettingsWithSourceTrace"})}),">"]})]})}function p(e={}){const{wrapper:t}={...(0,r.R)(),...e.components};return t?(0,s.jsx)(t,{...e,children:(0,s.jsx)(d,{...e})}):d(e)}}}]);