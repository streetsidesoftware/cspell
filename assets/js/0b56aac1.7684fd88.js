"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[16718],{17322:(e,s,i)=>{i.r(s),i.d(s,{assets:()=>o,contentTitle:()=>a,default:()=>p,frontMatter:()=>r,metadata:()=>l,toc:()=>c});const l=JSON.parse('{"id":"api/cspell-lib/variables/ConfigFields","title":"Variable: ConfigFields","description":"const ConfigFields: CSpellUserSettingsFields","source":"@site/docs/api/cspell-lib/variables/ConfigFields.md","sourceDirName":"api/cspell-lib/variables","slug":"/api/cspell-lib/variables/ConfigFields","permalink":"/docs/api/cspell-lib/variables/ConfigFields","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell-lib/variables/ConfigFields.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Type Alias: ~~VersionLegacy~~","permalink":"/docs/api/cspell-lib/type-aliases/VersionLegacy"},"next":{"title":"Variable: ENV\\\\_CSPELL\\\\_GLOB\\\\_ROOT","permalink":"/docs/api/cspell-lib/variables/ENV_CSPELL_GLOB_ROOT"}}');var t=i(37557),n=i(46039);const r={},a="Variable: ConfigFields",o={},c=[];function d(e){const s={a:"a",blockquote:"blockquote",code:"code",h1:"h1",header:"header",p:"p",strong:"strong",...(0,n.R)(),...e.components};return(0,t.jsxs)(t.Fragment,{children:[(0,t.jsx)(s.header,{children:(0,t.jsx)(s.h1,{id:"variable-configfields",children:"Variable: ConfigFields"})}),"\n",(0,t.jsxs)(s.blockquote,{children:["\n",(0,t.jsxs)(s.p,{children:[(0,t.jsx)(s.code,{children:"const"})," ",(0,t.jsx)(s.strong,{children:"ConfigFields"}),": ",(0,t.jsx)(s.a,{href:"/docs/api/cspell-lib/type-aliases/CSpellUserSettingsFields",children:(0,t.jsx)(s.code,{children:"CSpellUserSettingsFields"})})]}),"\n"]}),"\n",(0,t.jsx)(s.p,{children:"Defined in: packages/cspell-types/dist/index.d.mts:1566"})]})}function p(e={}){const{wrapper:s}={...(0,n.R)(),...e.components};return s?(0,t.jsx)(s,{...e,children:(0,t.jsx)(d,{...e})}):d(e)}},46039:(e,s,i)=>{i.d(s,{R:()=>r,x:()=>a});var l=i(18225);const t={},n=l.createContext(t);function r(e){const s=l.useContext(n);return l.useMemo((function(){return"function"==typeof e?e(s):{...s,...e}}),[s,e])}function a(e){let s;return s=e.disableParentContext?"function"==typeof e.components?e.components(t):e.components||t:r(e.components),l.createElement(n.Provider,{value:s},e.children)}}}]);