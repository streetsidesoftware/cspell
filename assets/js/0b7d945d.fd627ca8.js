"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[43752],{46039:(e,s,t)=>{t.d(s,{R:()=>o,x:()=>i});var n=t(18225);const r={},c=n.createContext(r);function o(e){const s=n.useContext(c);return n.useMemo((function(){return"function"==typeof e?e(s):{...s,...e}}),[s,e])}function i(e){let s;return s=e.disableParentContext?"function"==typeof e.components?e.components(r):e.components||r:o(e.components),n.createElement(c.Provider,{value:s},e.children)}},69483:(e,s,t)=>{t.r(s),t.d(s,{assets:()=>l,contentTitle:()=>i,default:()=>p,frontMatter:()=>o,metadata:()=>n,toc:()=>a});const n=JSON.parse('{"id":"api/cspell/functions/trace","title":"Function: trace()","description":"trace(words, options): AsyncIterableIterator\\\\","source":"@site/docs/api/cspell/functions/trace.md","sourceDirName":"api/cspell/functions","slug":"/api/cspell/functions/trace","permalink":"/docs/api/cspell/functions/trace","draft":false,"unlisted":false,"editUrl":"https://github.com/streetsidesoftware/cspell/tree/main/website/docs/docs/api/cspell/functions/trace.md","tags":[],"version":"current","frontMatter":{},"sidebar":"tutorialSidebar","previous":{"title":"Function: suggestions()","permalink":"/docs/api/cspell/functions/suggestions"},"next":{"title":"cspell","permalink":"/docs/api/cspell/globals"}}');var r=t(37557),c=t(46039);const o={},i="Function: trace()",l={},a=[{value:"Parameters",id:"parameters",level:2},{value:"words",id:"words",level:3},{value:"options",id:"options",level:3},{value:"Returns",id:"returns",level:2}];function d(e){const s={a:"a",blockquote:"blockquote",code:"code",h1:"h1",h2:"h2",h3:"h3",header:"header",p:"p",strong:"strong",...(0,c.R)(),...e.components};return(0,r.jsxs)(r.Fragment,{children:[(0,r.jsx)(s.header,{children:(0,r.jsx)(s.h1,{id:"function-trace",children:"Function: trace()"})}),"\n",(0,r.jsxs)(s.blockquote,{children:["\n",(0,r.jsxs)(s.p,{children:[(0,r.jsx)(s.strong,{children:"trace"}),"(",(0,r.jsx)(s.code,{children:"words"}),", ",(0,r.jsx)(s.code,{children:"options"}),"): ",(0,r.jsx)(s.code,{children:"AsyncIterableIterator"}),"<",(0,r.jsx)(s.code,{children:"TraceWordResult"}),">"]}),"\n"]}),"\n",(0,r.jsxs)(s.p,{children:["Defined in: ",(0,r.jsx)(s.a,{href:"https://github.com/streetsidesoftware/cspell/blob/3fc3ef2ea9def847ef20287c6238ff423e6493eb/packages/cspell/src/app/application.mts#L44",children:"cspell/src/app/application.mts:44"})]}),"\n",(0,r.jsx)(s.h2,{id:"parameters",children:"Parameters"}),"\n",(0,r.jsx)(s.h3,{id:"words",children:"words"}),"\n",(0,r.jsxs)(s.p,{children:[(0,r.jsx)(s.code,{children:"string"}),"[]"]}),"\n",(0,r.jsx)(s.h3,{id:"options",children:"options"}),"\n",(0,r.jsx)(s.p,{children:(0,r.jsx)(s.a,{href:"/docs/api/cspell/interfaces/TraceOptions",children:(0,r.jsx)(s.code,{children:"TraceOptions"})})}),"\n",(0,r.jsx)(s.h2,{id:"returns",children:"Returns"}),"\n",(0,r.jsxs)(s.p,{children:[(0,r.jsx)(s.code,{children:"AsyncIterableIterator"}),"<",(0,r.jsx)(s.code,{children:"TraceWordResult"}),">"]})]})}function p(e={}){const{wrapper:s}={...(0,c.R)(),...e.components};return s?(0,r.jsx)(s,{...e,children:(0,r.jsx)(d,{...e})}):d(e)}}}]);