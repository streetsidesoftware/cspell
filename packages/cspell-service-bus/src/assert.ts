export function assert(value: unknown, message?: string | Error): asserts value {
    if (!value) {
        const err = message instanceof Error ? message : Error(message ?? 'AssertionError');
        throw err;
    }
}
