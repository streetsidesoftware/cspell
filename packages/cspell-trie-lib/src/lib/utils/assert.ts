export function assert(condition: unknown, message = 'Assert Failed'): asserts condition {
    if (condition) return;
    throw new Error(message);
}
