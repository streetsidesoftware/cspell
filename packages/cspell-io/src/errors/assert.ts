import { AssertionError } from './errors.js';

export function assert(value: unknown, message?: string): asserts value {
    if (!value) {
        throw new AssertionError(message ?? 'Assertion failed');
    }
}
