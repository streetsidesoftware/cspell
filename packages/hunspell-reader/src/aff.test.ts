import assert from 'assert';
import { readdirSync } from 'fs';
import * as path from 'path';
import { describe, expect, it } from 'vitest';

import { Aff, affWordToColoredString, asAffWord, compareAff, filterAff, flagsToString } from './aff.js';
import type { AffWord } from './affDef.js';
import * as AffReader from './affReader.js';
import { parseAffFileToAff } from './affReader.js';

const isLoggerOn = false;
const DICTIONARY_LOCATIONS = path.join(__dirname, '..', 'dictionaries');
const nlAff = path.join(DICTIONARY_LOCATIONS, 'nl.aff');
const enAff = path.join(DICTIONARY_LOCATIONS, 'en_US.aff');
// const enGbAff = path.join(DICTIONARY_LOCATIONS, 'en_GB.aff');
const esAff = path.join(DICTIONARY_LOCATIONS, 'es_ANY.aff');
const frAff = path.join(DICTIONARY_LOCATIONS, 'fr-moderne.aff');
const huAff = path.join(DICTIONARY_LOCATIONS, 'hu', 'hu.aff');

describe('Basic Aff Validation', () => {
    const pAff = AffReader.parseAff(getSimpleAff());
    it('Reads Simple Aff', async () => {
        const aff = await pAff;
        expect(aff.SET).toBe('UTF-8');
        expect(aff.PFX).toBeInstanceOf(Map);
        expect(aff.SFX).toBeInstanceOf(Map);
    });
    it('Checks the PFX values', async () => {
        const aff = await pAff;
        assert(aff.PFX);
        expect(aff.PFX).toBeInstanceOf(Map);
        expect(aff.PFX.has('X')).toBe(true);
        const fx = aff.PFX.get('X');
        expect(fx).toBeDefined();
    });
    it('Checks the SFX values', async () => {
        const aff = await pAff;
        assert(aff.SFX);
        expect(aff.SFX).toBeInstanceOf(Map);
        expect(aff.SFX.has('J')).toBe(true);
        const fxJ = aff.SFX.get('J');
        expect(fxJ).toBeDefined();
    });

    it('Checks ICONV OCONV', () => {
        const aff = new Aff(AffReader.parseAff(getSampleAffIconvOconv()));
        expect(aff.iConv.convert('abc')).toBe('abc');
        expect(aff.iConv.convert('ABC')).toBe('abc');
        expect(aff.iConv.convert('á Á')).toBe('á á');

        expect(aff.oConv.convert('abc')).toBe('ABC');
        expect(aff.oConv.convert('ABC')).toBe('ABC');
        expect(aff.oConv.convert('á Á')).toBe('Á Á');
    });
});

describe('Test Aff', () => {
    it('tests applying rules for fr `badger/10`', async () => {
        const aff = await parseAffFileToAff(frAff);
        const r = aff.applyRulesToDicEntry('badger/10');
        const w = r.map((affWord) => affWord.word);
        expect(w).toEqual(expect.arrayContaining(['badger']));
        expect(w).toEqual(expect.arrayContaining(['badgeant']));
        logApplyRulesResults(r);
    });

    it('tests applying rules for fr `avoir/180`', async () => {
        const aff = await parseAffFileToAff(frAff);
        const r = aff.applyRulesToDicEntry('avoir/180');
        const w = r.map((affWord) => affWord.word);
        expect(w).toEqual(expect.arrayContaining(['avoir']));
        expect(w).toEqual(expect.arrayContaining(['n’avoir'])); // cspell:ignore n’avoir
        expect(r.map((affW) => affW.word)).toEqual(
            r.map((affW) => aff.oConv.convert(affW.prefix + affW.base + affW.suffix))
        );
        logApplyRulesResults(r);
    });

    it('tests applying rules for fr with maxDepth', async () => {
        const aff = await parseAffFileToAff(frAff);
        aff.maxSuffixDepth = 1;
        const r0 = aff.applyRulesToDicEntry('avoir/180').map((affWord) => affWord.word);
        expect(r0).toEqual(expect.arrayContaining(['avoir']));
        expect(r0).toEqual(expect.not.arrayContaining(['n’avoir'])); // cspell:ignore n’avoir
        aff.maxSuffixDepth = 2;
        const r1 = aff.applyRulesToDicEntry('avoir/180').map((affWord) => affWord.word);
        expect(r1).toEqual(expect.arrayContaining(['avoir']));
        expect(r1).toEqual(expect.arrayContaining(['n’avoir'])); // cspell:ignore n’avoir
        expect(r1).not.toEqual(r0);
        const r2 = aff.applyRulesToDicEntry('avoir/180', 1).map((affWord) => affWord.word);
        expect(r2).toEqual(r0);
    });

    it('test breaking up rules for nl', async () => {
        const aff = await parseAffFileToAff(nlAff);
        expect(aff.separateRules('ZbCcChC1')).toEqual(['Zb', 'Cc', 'Ch', 'C1']);
        expect(aff.separateRules('ZbCcChC199')).toEqual(['Zb', 'Cc', 'Ch', 'C1', '99']);
    });

    it('test breaking up rules for en', async () => {
        const aff = await parseAffFileToAff(enAff);
        expect(aff.separateRules('ZbCcChC1')).not.toEqual(['Zb', 'Cc', 'Ch', 'C1']);
        expect(aff.separateRules('ZbCcChC1')).toEqual('ZbCch1'.split(''));
    });

    it('test getting rules for nl', async () => {
        const aff = await parseAffFileToAff(nlAff);
        // console.log(aff.getMatchingRules('ZbCcChC1'));
        expect(
            aff
                .getMatchingRules('ZbCcChC1')
                .filter((a) => !!a)
                .map(({ id }) => id)
        ).toEqual(['Zb', 'Cc', 'Ch']);
        expect(
            aff
                .getMatchingRules('ZbCcChC199')
                .filter((a) => !!a)
                .map(({ id }) => id)
        ).toEqual(['Zb', 'Cc', 'Ch']);
        expect(
            aff
                .getMatchingRules('AaAbAcAdAeAi')
                .filter((a) => !!a)
                .map(({ id }) => id)
        ).toEqual(['Aa', 'Ab', 'Ac', 'Ad', 'Ae', 'Ai']);
        expect(
            aff
                .getMatchingRules('AaAbAcAdAeAi')
                .filter((a) => !!a)
                .map(({ type }) => type)
        ).toEqual(['sfx', 'sfx', 'sfx', 'sfx', 'sfx', 'sfx']);
        expect(
            aff
                .getMatchingRules('PaPbPc')
                .filter((a) => !!a)
                .map(({ type }) => type)
        ).toEqual(['pfx', 'pfx', 'pfx']);
    });

    it('tests applying rules for nl', async () => {
        const aff = await parseAffFileToAff(nlAff);
        const lines = ['dc/ClCwKc', 'aak/Zf', 'huis/CACcYbCQZhC0', 'pannenkoek/ZbCACcC0'];
        const appliedRules = lines.map((line) => aff.applyRulesToDicEntry(line).map(formatAffWordForSnapshot));
        expect(appliedRules).toMatchSnapshot();
    });

    it('tests applying rules for es', async () => {
        const aff = await parseAffFileToAff(esAff);
        const lines = ['ababillar/RED'];
        const appliedRules = lines.map((line) => aff.applyRulesToDicEntry(line).map(formatAffWordForSnapshot));
        expect(appliedRules).toMatchSnapshot();
    });

    it('tests applying rules for en', async () => {
        const aff = await parseAffFileToAff(enAff);
        const r = aff.applyRulesToDicEntry('motivate/CDSG');
        const w = r.map((affWord) => affWord.word);
        expect(w.sort()).toEqual([
            'demotivate',
            'demotivated',
            'demotivates',
            'demotivating',
            'motivate',
            'motivated',
            'motivates',
            'motivating',
        ]);
    });

    it('tests compareAff', () => {
        expect(compareAff(asAffWord('word'), asAffWord('word'))).toBe(0);
        expect(compareAff(asAffWord('a word'), asAffWord('b word'))).toBe(-1);
        expect(compareAff(asAffWord('b word'), asAffWord('a word'))).toBe(1);
        const affA = asAffWord('word');
        const affB = asAffWord('word');
        affA.flags.isCompoundPermitted = true;
        expect(compareAff(affA, affB)).toBe(1);
        affB.flags.isCompoundPermitted = true;
        expect(compareAff(affA, affB)).toBe(0);
        affB.flags.canBeCompoundBegin = true;
        expect(compareAff(affA, affB)).toBe(1);
    });

    it('test filterAff', () => {
        const fn = filterAff();
        expect(fn(asAffWord('Hello'))).toBe(true);
        expect(fn(asAffWord('Hello'))).toBe(false);
        expect(fn(asAffWord('Hello', '', { canBeCompoundBegin: true }))).toBe(true);
        expect(fn(asAffWord('Hello', '', { canBeCompoundBegin: true }))).toBe(false);
        expect(fn(asAffWord('Hello'))).toBe(true);
        expect(fn(asAffWord('Hello'))).toBe(false);
        expect(fn(asAffWord('There'))).toBe(true);
        expect(fn(asAffWord('There'))).toBe(false);
    });
});

describe('Validated loading all dictionaries in the `dictionaries` directory.', () => {
    function getDictionaries() {
        return readdirSync(DICTIONARY_LOCATIONS)
            .filter((dic) => !!dic.match(/\.aff$/))
            .map((base) => path.join(DICTIONARY_LOCATIONS, base));
    }
    const dictionaries = getDictionaries();
    it('Make sure we found some sample dictionaries', () => {
        expect(dictionaries.length).toBeGreaterThan(4);
    });

    dictionaries.forEach((dicAff) => {
        // const dicDic = dicAff.replace(/\.aff$/, '.dic');
        // const dicContent = fs.readFile(dicDic)
        it(`Ensure we can load ${path.basename(dicAff)}`, async () => {
            const aff = await AffReader.parseAffFile(dicAff);
            expect(aff.PFX).toBeInstanceOf(Map);
            expect(aff.SFX).toBeInstanceOf(Map);
        });
    });
});

describe('Validate loading Hungarian', () => {
    const affP = parseAffFileToAff(huAff);

    it('tests applying rules for hu Depth 0', async () => {
        const aff = await affP;
        aff.maxSuffixDepth = 0;
        const r = aff.applyRulesToDicEntry('kemping/17');
        const w = r.map((affWord) => affWord.word);
        expect(w).toEqual(expect.arrayContaining(['kemping']));
        expect(w.length).toBe(1);
    });

    it('tests applying rules for hu Depth 1', async () => {
        const aff = await affP;
        aff.maxSuffixDepth = 1;
        const r = aff.applyRulesToDicEntry('kemping/17');
        const w = r.map((affWord) => affWord.word);
        expect(w).toEqual(expect.arrayContaining(['kemping']));
        expect(w.length).toBeGreaterThan(1);
    });
});

function formatAffWordForSnapshot(affWord: AffWord): string {
    const { dic, base, rulesApplied, word, suffix, prefix, flags } = affWord;
    const f = flagsToString(flags);
    return `${dic} -> ${base} (${rulesApplied.trim()}) [${prefix}|${suffix}] --> '${word}${f ? '/' + f : ''}'`;
}

function logApplyRulesResults(affWords: AffWord[]) {
    affWords.forEach(logApplyRulesResult);
}

function logApplyRulesResult(affWord: AffWord) {
    if (isLoggerOn) console.log(affWordToColoredString(affWord));
}

function getSimpleAff() {
    // cspell:ignore esianrtolcdugmphbyfvkwzESIANRTOLCDUGMPHBYFVKWZ
    return `
    SET UTF-8
    TRY esianrtolcdugmphbyfvkwzESIANRTOLCDUGMPHBYFVKWZ'
    ICONV 1
    ICONV ’ '
    NOSUGGEST !

    # ordinal numbers
    COMPOUNDMIN 1
    # only in compounds: 1th, 2th, 3th
    ONLYINCOMPOUND c
    # compound rules:
    # 1. [0-9]*1[0-9]th (10th, 11th, 12th, 56714th, etc.)
    # 2. [0-9]*[02-9](1st|2nd|3rd|[4-9]th) (21st, 22nd, 123rd, 1234th, etc.)
    COMPOUNDRULE 2
    COMPOUNDRULE n*1t
    COMPOUNDRULE n*mp
    WORDCHARS 0123456789

    PFX A Y 1
    PFX A   0     re         .

    PFX I Y 1
    PFX I   0     in         .

    PFX U Y 1
    PFX U   0     un         .

    PFX X Y 1
    PFX X   0     un         .
    PFX X   0     re         .
    PFX X   0     in         .
    PFX X   0     a          .

    SFX Y Y 2
    SFX Y   0     ly         [^y]
    SFX Y   y     ily        [y]

    SFX G Y 2
    SFX G   e     ing        e
    SFX G   0     ing        [^e]

    SFX J Y 2
    SFX J   e     ings       e
    SFX J   0     ings       [^e]
    `;
}

function getSampleAffIconvOconv() {
    return `
# input output conversion
SET UTF-8

OCONV 7
OCONV a A
OCONV á Á
OCONV b B
OCONV c C
OCONV d D
OCONV e E
OCONV é É

ICONV 7
ICONV A a
ICONV Á á
ICONV B b
ICONV C c
ICONV D d
ICONV E e
ICONV É é

    `;
}

// cspell:ignore moderne avoir huis pannenkoek ababillar CDSG ings
// cspell:enableCompoundWords
