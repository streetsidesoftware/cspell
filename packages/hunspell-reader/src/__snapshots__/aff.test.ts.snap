// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Aff > tests applying rules for es 1`] = `
[
  [
    "'ababilla' Rules: E line: ababillar/RED",
    "'ababillaba' Rules: R line: ababillar/RED",
    "'ababillabais' Rules: R line: ababillar/RED",
    "'ababillaban' Rules: R line: ababillar/RED",
    "'ababillabas' Rules: R line: ababillar/RED",
    "'ababillad' Rules: R line: ababillar/RED",
    "'ababillada' Rules: D line: ababillar/RED",
    "'ababilladas' Rules: D line: ababillar/RED",
    "'ababillado' Rules: D line: ababillar/RED",
    "'ababillados' Rules: D line: ababillar/RED",
    "'ababillamos' Rules: R line: ababillar/RED",
    "'ababillan' Rules: E line: ababillar/RED",
    "'ababillando' Rules: R line: ababillar/RED",
    "'ababillar'",
    "'ababillara' Rules: R line: ababillar/RED",
    "'ababillarais' Rules: R line: ababillar/RED",
    "'ababillaran' Rules: R line: ababillar/RED",
    "'ababillaras' Rules: R line: ababillar/RED",
    "'ababillare' Rules: R line: ababillar/RED",
    "'ababillareis' Rules: R line: ababillar/RED",
    "'ababillaremos' Rules: R line: ababillar/RED",
    "'ababillaren' Rules: R line: ababillar/RED",
    "'ababillares' Rules: R line: ababillar/RED",
    "'ababillaron' Rules: R line: ababillar/RED",
    "'ababillarse' Rules: R line: ababillar/RED",
    "'ababillará' Rules: R line: ababillar/RED",
    "'ababillarán' Rules: R line: ababillar/RED",
    "'ababillarás' Rules: R line: ababillar/RED",
    "'ababillaré' Rules: R line: ababillar/RED",
    "'ababillaréis' Rules: R line: ababillar/RED",
    "'ababillaría' Rules: R line: ababillar/RED",
    "'ababillaríais' Rules: R line: ababillar/RED",
    "'ababillaríamos' Rules: R line: ababillar/RED",
    "'ababillarían' Rules: R line: ababillar/RED",
    "'ababillarías' Rules: R line: ababillar/RED",
    "'ababillas' Rules: E line: ababillar/RED",
    "'ababillase' Rules: R line: ababillar/RED",
    "'ababillaseis' Rules: R line: ababillar/RED",
    "'ababillasen' Rules: R line: ababillar/RED",
    "'ababillases' Rules: R line: ababillar/RED",
    "'ababillaste' Rules: R line: ababillar/RED",
    "'ababillasteis' Rules: R line: ababillar/RED",
    "'ababille' Rules: E line: ababillar/RED",
    "'ababillemos' Rules: R line: ababillar/RED",
    "'ababillen' Rules: E line: ababillar/RED",
    "'ababilles' Rules: E line: ababillar/RED",
    "'ababillo' Rules: E line: ababillar/RED",
    "'ababillá' Rules: R line: ababillar/RED",
    "'ababillábamos' Rules: R line: ababillar/RED",
    "'ababilláis' Rules: R line: ababillar/RED",
    "'ababillándose' Rules: R line: ababillar/RED",
    "'ababilláramos' Rules: R line: ababillar/RED",
    "'ababilláremos' Rules: R line: ababillar/RED",
    "'ababillás' Rules: R line: ababillar/RED",
    "'ababillásemos' Rules: R line: ababillar/RED",
    "'ababillé' Rules: R line: ababillar/RED",
    "'ababilléis' Rules: R line: ababillar/RED",
    "'ababilló' Rules: R line: ababillar/RED",
  ],
]
`;

exports[`Test Aff > tests applying rules for nl 1`] = `
[
  [
    "'-dc/CompoundEnd:CompoundMiddle:CompoundPermitted:KeepCase' Rules: Kc,Cw,Cb,Cc,Cp line: dc/ClCwKc",
    "'-dc-/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted:KeepCase' Rules: Kc,Cw,Cb,Cc,Cp,Cl,Ca,Cb,Cp line: dc/ClCwKc",
    "'dc/KeepCase' Rules: Kc line: dc/ClCwKc",
    "'dc-/CompoundBegin:CompoundMiddle:CompoundPermitted:KeepCase' Rules: Kc,Cl,Ca,Cb,Cp line: dc/ClCwKc",
  ],
  [
    "'aak'",
    "'aken/CompoundBegin:CompoundMiddle:CompoundPermitted' Rules: Zf,Ca,Cb,Cp line: aak/Zf",
    "'aken-/CompoundBegin:CompoundMiddle:CompoundPermitted' Rules: Zf,Ca,Cb,Cp line: aak/Zf",
  ],
  [
    "'huis/CompoundEnd' Rules: Cc line: huis/CACcYbCQZhC0",
    "'huis/CompoundBegin:CompoundEnd:CompoundPermitted' Rules: Cc,CA,Ca,Cp line: huis/CACcYbCQZhC0",
    "'huis-/CompoundBegin:CompoundEnd:CompoundPermitted' Rules: Cc,CA,Ca,Cp line: huis/CACcYbCQZhC0",
    "'huisje/CompoundEnd' Rules: Cc,Yb line: huis/CACcYbCQZhC0",
    "'huisjes/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted' Rules: Cc,Yb,Ca,Cb,Cp line: huis/CACcYbCQZhC0",
    "'huisjes-/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted' Rules: Cc,Yb,Ca,Cb,Cp line: huis/CACcYbCQZhC0",
    "'huizen/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted' Rules: Cc,Zh,Ca,Cb,Cp line: huis/CACcYbCQZhC0",
    "'huizen-/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted' Rules: Cc,Zh,Ca,Cb,Cp line: huis/CACcYbCQZhC0",
  ],
  [
    "'pannenkoek/CompoundEnd' Rules: Cc line: pannenkoek/ZbCACcC0",
    "'pannenkoek/CompoundBegin:CompoundEnd:CompoundPermitted' Rules: Cc,CA,Ca,Cp line: pannenkoek/ZbCACcC0",
    "'pannenkoek-/CompoundBegin:CompoundEnd:CompoundPermitted' Rules: Cc,CA,Ca,Cp line: pannenkoek/ZbCACcC0",
    "'pannenkoeken/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted' Rules: Cc,Zb,Ca,Cb,Cp line: pannenkoek/ZbCACcC0",
    "'pannenkoeken-/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted' Rules: Cc,Zb,Ca,Cb,Cp line: pannenkoek/ZbCACcC0",
  ],
]
`;

exports[`Test Aff > tests applying rules for nl huis 1`] = `
[
  "'huis/CompoundEnd' Rules: Cc line: huis/CACcYbCQZhC0",
  "'huis/CompoundBegin:CompoundEnd:CompoundPermitted' Rules: Cc,CA,Ca,Cp line: huis/CACcYbCQZhC0",
  "'huis-/CompoundBegin:CompoundEnd:CompoundPermitted' Rules: Cc,CA,Ca,Cp line: huis/CACcYbCQZhC0",
  "'huisje/CompoundEnd' Rules: Cc,Yb line: huis/CACcYbCQZhC0",
  "'huisjes/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted' Rules: Cc,Yb,Ca,Cb,Cp line: huis/CACcYbCQZhC0",
  "'huisjes-/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted' Rules: Cc,Yb,Ca,Cb,Cp line: huis/CACcYbCQZhC0",
  "'huizen/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted' Rules: Cc,Zh,Ca,Cb,Cp line: huis/CACcYbCQZhC0",
  "'huizen-/CompoundBegin:CompoundEnd:CompoundMiddle:CompoundPermitted' Rules: Cc,Zh,Ca,Cb,Cp line: huis/CACcYbCQZhC0",
]
`;
